// sign_module/lib/main.js
module.exports = start;

function start(config){
    var app = config.app;

    app.get('/signPage', function(req, res){
        res.sendFile(config.directory + '/signPage/SignIn.html');
    });

    app.get('/signPage/signIn', function(req, res){
        res.sendFile(config.directory + '/index.html');
    });

    app.get('/signPage/signUP', function(req, res){
        res.sendFile(config.directory + '/index.html');
    });

    app.get('/signPage/logOut', function(req, res){
        res.sendFile(config.directory + '/index.html');
    });

    return (new API(config));
};

function signIn(){

}

function signUp(name,password){
    var sql_string = 'SELECT Name FROM PersonalInformation WHERE Name=' + name;

    config.modules['saferman'].sql(sql_string,function(results){
        if(results[0] != undefined){
            //insert data into PersonalInformation
            sql_string = 'INSERT INTO PersonalInformation (ID,Name,Graph,PhoneNumber,Score) VALUES (0,hehe,0,0,0)'
            config.modules['saferman'].sql(sql_string);

            //insert data into ShadowTable
            sql_string = 'INSERT INTO PersonalInformation (ID,Name,Graph,PhoneNumber,Score) VALUES (0,hehe,0,0,0)'
            config.modules['saferman'].sql(sql_string);

            //insert data into rightsTable
            sql_string = 'INSERT INTO PersonalInformation (ID,Name,Graph,PhoneNumber,Score) VALUES (0,hehe,0,0,0)'
            config.modules['saferman'].sql(sql_string);
        }else{
            //return false
        }
    });
}

function logOut(){

}

function API(config){
    this.config = config;
}

API.prototype.hehe = function(){
    console.log('asdf');
}
